otelcol/collector_core.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelcol // import "go.opentelemetry.io/collector/otelcol"
+
+import (
+	"sync"
+
+	"go.uber.org/zap/zapcore"
+)
+
+var _ zapcore.Core = (*collectorCore)(nil)
+
+type collectorCore struct {
+	core zapcore.Core
+	mu   sync.Mutex
```

Comment:
Would a `RWMutex` make more sense here? (I am actually not sure). 

My reasoning: `core` must already guarantee that you can call its methods concurrently, the only thing we have to guard against is updating the field itself. It should be safe to `Lock` in `SetCore`, but `RLock` in any other method.

Reply:
I think you're right